Sources/BraveWallet/Crypto/BuySendSwap/SendTokenView.swift

```diff
@@ -24,20 +24,24 @@ struct SendTokenView: View {
   var onDismiss: () -> Void
 
   private var isSendDisabled: Bool {
-    guard let sendAmount = BDouble(sendTokenStore.sendAmount.normalizedDecimals),
-      let balance = sendTokenStore.selectedSendTokenBalance,
-      let token = sendTokenStore.selectedSendToken,
-      !sendTokenStore.isMakingTx
-    else {
+    guard let balance = sendTokenStore.selectedSendTokenBalance,
+          let token = sendTokenStore.selectedSendToken,
+          !sendTokenStore.isMakingTx,
+          !sendTokenStore.sendAddress.isEmpty,
+          sendTokenStore.addressError == nil else {
+      return true
+    }
+    if token.isErc721 || token.isNft {
+      return balance < 1
```

Comment:
do we want to align with Desktop to have a `insufficient balance` error message for NFTs?

Reply:
This exposed a bad race condition where we were showing assets and the selected asset balance for the wrong network:
![SendTokenStore - race condition](https://user-images.githubusercontent.com/5314553/201945325-94942cd8-86ed-4937-b8a7-c44c969aac01.jpeg)

In addition to adding the `insufficient balance` error to the send button (to match swap), I've reworked fetching assets & balances to resolve the race condition in [acb28c6](https://github.com/brave/brave-ios/pull/6385/commits/acb28c645852958840b766973a510e33240b2789). `fetchAssets` and `fetchBalances` are combined into `update()` now which will fetch assets and the balance for the selected asset, then verify that balance. Please test switching networks and verify the assets are correct. Creating transactions remains the same, except for `sendToken()` which will fetch the selected account and pass to eth/sol functions to create the transactions.